olive/data_config/container/huggingface_container.py

```diff
@@ -0,0 +1,26 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+from typing import ClassVar
+
+from olive.data_config.constants import DataComponentType, DataContainerType
+from olive.data_config.container.base_container import BaseContainer
+from olive.data_config.registry import Registry
+
+
+@Registry.register(DataContainerType.DATA_CONTAINER)
+class HuggingfaceContainer(BaseContainer):
```

Comment:
How about organize the code structure like:

data_config = {
    "default": {
        "dataset": local_dataset,
        "pre_process": pre_process,
        "post_process": post_process,
        "dataloader": default_dataloader,
    },
    "huggingface": {
        "dataset": huggingface_dataset,
        "pre_process": huggingface_pre_process,
        "task_type_components_map": {
            "post_process": text_classification_post_process
        }
    }
}

In this way, there only need
* config.py to handle the configuration logic
* default.py, which include all the default dataloader/dataset/pre_proces/post_process logic
* huggingface.py, which include all the dataset/pre_process/post_process logic for hugginface dataset.
* dataset.py, it is the rename of existing base_container.py, which include the logic to invoke the dataset/dataloader etc.

All other files including registry.py, constant.py could be removed. At the same time, the organization seems neutral than current one since all the dataset related components like dataloader/pt_dataset/pre_process/post_process is put into a single file instead of spreading into multiple files. 

How do you think?

Reply:
In this way, user could easily extend the dataset by providing my_dataset.py

```python
def my_dataset(data_dir=None, label_cols=None, **kwargs):
    pass

def my_pre_process(dataset):
    pass

def my_post_process(output_data, **kwargs):
    pass

my_data_config = {
    "dataset": my_dataset,
    "pre_process": my_pre_process,
    "post_process": my_post_process,
}

data_config["my_dataset"] = my_data_config
```

Reply:
Thanks for your comments. We may want to keep current registry module to support components combinations for customers.